File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/convert/Converter.java
/*
 * Copyright 2016 jeasonlzy(廖子尧)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.ashlikun.okhttputils.http.convert;

import com.google.gson.Gson;

import okhttp3.Response;

/**
 * @author　　: 李坤
 * 创建时间: 2018/5/10 0010 上午 10:30
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：网络数据的转换接口
 */
public interface Converter<T> {

    /**
     * 拿到响应后，将数据转换成需要的格式，子线程中执行，可以是耗时操作
     *
     * @param response 需要转换的对象
     * @param gosn     gson解析
     * @return 转换后的结果
     * @throws Exception 转换过程发生的异常
     */
    T convertResponse(Response response, Gson gosn) throws Exception;
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/convert/Converter.kt
package com.ashlikun.okhttputils.http.convert

import com.google.gson.Gson
import okhttp3.Response

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.16 14:13
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：网络数据的转换接口
 */

interface Converter<T> {
    /**
     * 拿到响应后，将数据转换成需要的格式，子线程中执行，可以是耗时操作
     *
     * @param response 需要转换的对象
     * @param gosn     gson解析
     * @return 转换后的结果
     * @throws Exception 转换过程发生的异常
     */
    @Throws(Exception::class)
    fun convertResponse(response: Response, gosn: Gson): T
}